ide/git/src/org/netbeans/modules/git/ui/history/DefaultGitHyperlinkProvider.java

```diff
@@ -0,0 +1,126 @@
+
+/**
+ * Looks for '#123' patterns in text (e.g commit messages) and tries to link
+ * them to a fitting web page.
+ * 
+ * <p>The repository host is identified by inspecting the registered remotes
+ * of the local git repository.
+ * 
+ * @author mbien
+ */
+@Messages({
+     "# {0} - issue ID",
+     "TT_DefaultGitHyperlinkProvider.link=Click to open {0} in Browser",
+     "MSG_DefaultGitHyperlinkProvider.link.failed=Could not resolve issue link."
+ })
+@ServiceProvider(service=VCSHyperlinkProvider.class, position = Integer.MAX_VALUE)
+public class DefaultGitHyperlinkProvider extends VCSHyperlinkProvider {
+    
+    private static final Pattern ids = Pattern.compile("#[0-9]+");
+
+    @Override
+    public int[] getSpans(String text) {
+        return ids.matcher(text).results()
+                                .flatMapToInt(r -> IntStream.of(r.start(), r.end()))
+                                .toArray();
+    }
+
+    @Override
+    public String getTooltip(String text, int offsetStart, int offsetEnd) {
+        return Bundle.TT_DefaultGitHyperlinkProvider_link(text.substring(offsetStart, offsetEnd));
+    }
+
+    @Override
+    public void onClick(File file, String text, int offsetStart, int offsetEnd) {
+        Path path = file.toPath();
+        while (!Files.isDirectory(path) || Files.notExists(path.resolve(".git"))) {
+            path = path.getParent();
+            if (path == null) {
+                return;
+            }
+        }
+        Path repo = path;
+        RequestProcessor.getDefault().post(() -> {
+            openIssue(repo, text.substring(offsetStart + 1, offsetEnd));
+        });
+    }
+
+    private static void openIssue(Path repo, String id) {
+        try (GitClient client = GitRepository.getInstance(repo.toFile()).createClient()) {
+            Map<String, GitRemoteConfig> remotes = client.getRemotes(new DefaultProgressMonitor());
+            GitRemoteConfig origin = remotes.get("origin");
+            if (origin == null) {
+                origin = remotes.values().stream()
+                                         .filter(r -> r.getRemoteName().startsWith("origin/"))
+                                         .findFirst().orElse(null);
+            }
+            if (origin != null) {
+                for (String str : origin.getUris()) {
+                    GitURI uri = new GitURI(str);
+                    String path = uri.getPath().substring(0, uri.getPath().length()-4); // remove .git postfix
+                    if (!path.startsWith("/")) {
+                        path = "/" + path;
+                    }
+                    String page = null;
+                    if (uri.getHost().equals("github.com")) {
+                        page = "/pull/" + id; // url works for issues too
+                    } else if (uri.getHost().equals("gitlab.com")) {
+                        page = "/-/issues/" + id; // does gitlab auto link merge_requests ?
+                    }
```

Comment:
At a later time (if anyone bothers to ask) we could add an option to add hostnames for systems to be treated as gitlab. My observation is, that it is not that hard to setup gitlab for local usage and thus is likely to be found in the wild.

Reply:
yes I agree. I actually tried to find other hosts but those two seem to be the most common ones. I have no bitbucket account so I don't know how those projects work.

Extending this is super easy though. Community can add more branches.

Reply:
Well, I'm working with upstream native code that uses `gitlab.freedesktop.org`.  Which makes me wonder how many people who do that put `gitlab` in the host?  What about checking with `uri.getHost().contains("gitlab")`?

Reply:
link me an example project first, ideally one which has issues in their history ;)

Reply:
nevermind, i found one on first attempt